lib/install.js

```diff
@@ -31,34 +34,29 @@ function install (gyp, argv, callback) {
     if (err) {
       log.warn('install', 'got an error, rolling back install')
       // roll-back the install if anything went wrong
-      gyp.commands.remove([ version ], function (err2) {
+      gyp.commands.remove([ release.version ], function (err2) {
         callback(err)
       })
     } else {
-      callback(null, version)
+      callback(null, release.version)
     }
   }
 
-  var distUrl = gyp.opts['dist-url'] || gyp.opts.disturl || 'https://nodejs.org/dist'
-
-
   // Determine which node dev files version we are installing
-  var versionStr = argv[0] || gyp.opts.target || process.version
-  log.verbose('install', 'input version string %j', versionStr)
+  log.verbose('install', 'input version string %j', release.version)
 
   // parse the version to normalize and ensure it's valid
-  var version = semver.parse(versionStr)
-  if (!version) {
-    return callback(new Error('Invalid version number: ' + versionStr))
+  if (!release.semver) {
+    return callback(new Error('Invalid version number: ' + release.version))
   }
 
-  if (semver.lt(versionStr, '0.8.0')) {
-    return callback(new Error('Minimum target version is `0.8.0` or greater. Got: ' + versionStr))
+  if (semver.lt(release.version, '0.8.0')) {
+    return callback(new Error('Minimum target version is `0.8.0` or greater. Got: ' + release.version))
   }
 
   // 0.x.y-pre versions are not published yet and cannot be installed. Bail.
-  if (version.prerelease[0] === 'pre') {
-    log.verbose('detected "pre" node version', versionStr)
+  if (release.semver.prerelease[0] === 'pre') {
```

Comment:
@Fishrock123 `-pre` matching comes from https://github.com/nodejs/node/blob/a338eb3c3e712fcaacdb475cc6f3378259a3d534/src/node_version.h#L19 i.e. you compiled this thing yourself, you should use `--nodedir` to point to it